pkg/prometheus/agent/statefulset.go

```diff
@@ -370,7 +370,7 @@ func makeStatefulSetSpec(
 				InitContainers:               initContainers,
 				SecurityContext:              cpf.SecurityContext,
 				ServiceAccountName:           cpf.ServiceAccountName,
-				AutomountServiceAccountToken: ptr.To(true),
+				AutomountServiceAccountToken: cpf.AutomountServiceAccountToken,
```

Comment:
I expect that we preserve the existing behavior when the field isn't defined.
```suggestion
				AutomountServiceAccountToken: ptr.Deref(cpf.AutomountServiceAccountToken, true),
```

Reply:
not sure if it works 😅  will result in the following error (not that familiar with Go) :

```
cannot use ptr.Deref(cpf.AutomountServiceAccountToken, true) (value of type bool) as *bool value in struct literalcompiler[IncompatibleAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#IncompatibleAssign)
```

Got inspired by the spec setup from Alertmanager https://github.com/prometheus-operator/prometheus-operator/blob/e3f9f9269baf61d344c176a3ddff767d7c425e6c/pkg/alertmanager/statefulset.go#L783  where if omitted it defaults to true. Let me know if it still needs patching 😄

Reply:
@simonpasquier thank you for the suggestion, patched ✅